R/fortify-map.R

```diff
@@ -80,7 +86,23 @@ fortify.map <- function(model, data, ...) {
 map_data <- function(map, region = ".", exact = FALSE, ...) {
   check_installed("maps", reason = "for `map_data()`.")
   map_obj <- maps::map(map, region, exact = exact, plot = FALSE, fill = TRUE, ...)
-  fortify(map_obj)
+
+  if (!inherits(map_obj, "map")) {
+    return(fortify(map_obj))
+  }
```

Comment:
Do you know when `maps::map()` returns a non-`map` object? Maybe we should just ignore, or raise an error rather than forwarding to `fortify()`?

``` r
map <- c("county", "france", "italy", "nz", "state", "usa", "world", "world2")
vapply(
  map,
  \(x) class(maps::map(x, region = ".", exact = FALSE, plot = FALSE, fill = TRUE)),
  FUN.VALUE = character(1L)
)
#> county france  italy     nz  state    usa  world world2 
#>  "map"  "map"  "map"  "map"  "map"  "map"  "map"  "map"
```

<sup>Created on 2024-12-10 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Reply:
If a user passes the `namesonly = TRUE` option, the `maps::map()` function does not return a non-map object.
Currently we throw the default error message, which will continue to be the case with this PR.

``` r
x <- maps::map(namesonly = TRUE, plot = FALSE)
str(x)
#>  chr [1:1627] "Aruba" "Afghanistan" "Angola" "Angola:Cabinda" "Anguilla" ...

ggplot2::map_data("world", namesonly = TRUE)
#> Error in `fortify()`:
#> ! `data` must be a <data.frame>, or an object coercible by `fortify()`,
#>   or a valid <data.frame>-like object coercible by `as.data.frame()`.
#> Caused by error in `.prevalidate_data_frame_like_object()`:
#> ! `dim(data)` must return an <integer> of length 2.
```

<sup>Created on 2024-12-10 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Reply:
Oh, I didn't know this, thanks.

> Happy to throw a more custom error message if you think that is more appropriate!

IMHO, throwing an error is preferable because we can avoid unnecessary `fortify()`. 

How about something like `` `maps::map()` didn't return a map object. Did you pass some unusual options to `map_data()`?  ``? I think this is a corner case, so I don't think it needs to be very user-friendly.

Reply:
Alright, now throwing the following error:

``` r
devtools::load_all("~/packages/ggplot2/")
#> ℹ Loading ggplot2
map_data("world", namesonly = TRUE)
#> Error in `map_data()`:
#> ! `maps::map()` must return an object of type <map>, not a character
#>   vector.
#> ℹ Did you pass the right arguments?
```

<sup>Created on 2024-12-10 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Reply:
The error looks good to me!